src/hydra-queue-runner/build-remote.cc

```diff
@@ -244,22 +244,29 @@ void State::buildRemote(ref<Store> destStore,
         updateStep(ssSendingInputs);
 
         StorePathSet inputs;
-        BasicDerivation basicDrv(*step->drv);
-
-        for (auto & p : step->drv->inputSrcs)
-            inputs.insert(p);
-
-        for (auto & input : step->drv->inputDrvs) {
-            auto drv2 = localStore->readDerivation(input.first);
-            for (auto & name : input.second) {
-                if (auto i = get(drv2.outputs, name)) {
-                    auto outPath = i->path(*localStore, drv2.name, name);
-                    inputs.insert(*outPath);
-                    basicDrv.inputSrcs.insert(*outPath);
+        BasicDerivation basicDrv;
+        auto outputHashes = staticOutputHashes(*localStore, *step->drv);
+        if (auto maybeBasicDrv = step->drv->tryResolve(*localStore))
```

Comment:
Note to self: Reading through the code again, I'm actually wondering whether this isn't messing-up the build of non-ca derivations (because `tryResolve` changes the output paths, so applying it unconditionally might do some funny things). I'd expect this would cause a test failure (or a hydra crash), but it's worth double-checking first